### PR TITLE
[SQL] [TEST] udf_java_method failed due to jdk version

### PR DESCRIPTION
**Pull request by scwf**
java.lang.Math.exp(1.0) has different result between jdk versions. so do not use createQueryTest, write a separate test for it.

```
jdk version     result 
1.7.0_11        2.7182818284590455  
1.7.0_05        2.7182818284590455
1.7.0_71        2.718281828459045
```

### CODE REVIEW

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveQuerySuite.scala [hunk 1]
```diff
@@ -291,6 +291,29 @@ class HiveQuerySuite extends HiveComparisonTest with BeforeAndAfter {
...
+        |FROM src tablesample (1 rows)
+      """.stripMargin).collect().head
+    Seq(
+      "1",
+      "true",
+      java.lang.Math.max(2, 3).toString,
+      java.lang.Math.min(2, 3).toString,
+      java.lang.Math.round(2.5).toString,
+      java.lang.Math.exp(1.0).toString,
+      java.lang.Math.floor(1.9).toString).zip(res.toSeq).foreach( x => assert(x._1 == x._2))
```
Comment (rxin) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveQuerySuite.scala]: can you remove the space before x?

Comment (scwf) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveQuerySuite.scala]: Done

**Comment by rxin** [comment]
cc @marmbrus - how do we want to handle tests like this?

**Comment by marmbrus** [comment]
Lets put it on the black list with a comment instead of just removing it from the white list so that it doesn't get added back by accident.  Lets also please use `checkAnswer` instead as it does better error reporting when there is a failure.

**Comment by marmbrus** [comment]
Thanks!  Merged to master and 1.4

**Pull request closed without merging**